qiskit/algorithms/minimum_eigen_solvers/minimum_eigen_solver.py

```diff
@@ -94,11 +98,11 @@ def eigenstate(self, value: np.ndarray) -> None:
         self._eigenstate = value
 
     @property
-    def aux_operator_eigenvalues(self) -> Optional[np.ndarray]:
+    def aux_operator_eigenvalues(self) -> Optional[ListOrDict[Union[float, complex]]]:
```

Comment:
```suggestion
    def aux_operator_eigenvalues(self) -> Optional[ListOrDict[complex]]:
```
I think `complex` implies `float`

Reply:
The same change can be done elsewhere

Reply:
Here I was a bit puzzled on the actual type of the eigenvalues. For the main operator it seems to be complex number but for the auxiliary operators it seem to be only real numbers.
After taking another look at `_eval_aux_operators` of `NumPyEigenSolver` and `_eval_aux_ops` of `VQE` it seems that the the first returns a `Tuple[float, float]` and the second return just a float.
Any suggestions which one to pick? Or should these both be cast to complex numbers?

Reply:
There is no need to cast them. You can pass a `float` to a function which expects `complex` because these are subtypes